src/IO.ts

```diff
@@ -1,6 +1,74 @@
 /**
- * @file `IO<A>` represents a synchronous computation that yields a value of type `A` and **never fails**.
- * If you want to represent a synchronous computation that may fail, please see `IOEither`.
+ * @file `IO<A>` represents a synchronous computation that yields a value of type `A` and **never fails**.  If you want
+ * to represent a synchronous computation that may fail, please see `IOEither`.  IO actions are terminated by calling
+ * their `run()` method that executes the computation and returns the result.
+ *
+ * ```ts
+ * import { IO, io } from 'fp-ts/lib/IO'
+ *
+ * const constant: IO<number> = io.of(123);
+ * constant.run()  // returns 123
+ *
+ * const random: IO<number> = new IO(() => Math.random())
+ * random.run()  // returns a random number
+ * random.run()  // returns another random number
+ * ```
+ *
+ * Below is a minimalistic example where `IO` datatype is used to implement a safe version of `console.log`.  Using
+ * `console.log` directly would break the code, resulting in both logging actions being executed when the value is not
+ * null.
+ *
+ * ```ts
+ * import { IO } from 'fp-ts/lib/IO'
+ * import { fromNullable } from 'fp-ts/lib/Option'
+ *
+ * const safeLog = (...args): IO<void> => new IO(() => console.log(...args));
+ *
+ * const logger = (input: number|null): IO<void> => fromNullable(input).fold(
+ *   safeLog('Received null'),
+ *   (value) => safeLog(`Received ${value}`),
+ * );
+ *
+ * logger(123).run();
+ * ```
+ *
+ * It is often convenient to combine several `IO` actions into one for sequential execution.
+ * You can use `sequence_(io, array)` to do this.
+ * ```ts
+ * import { IO, io } from 'fp-ts/lib/IO'
+ * import { array } from 'fp-ts/lib/Array'
+ * import { sequence_ } from 'fp-ts/lib/Foldable2v'
+ *
+ * const safeLog = (...args): IO<void> => new IO(() => console.log(...args));
+ *
+ * const logFoo: IO<void> = safeLog('foo');
```

Comment:
If you can think of a more realistic example instead of "foo" and "bar" that would be nice (some people don't know what "foo" means). If not, that is fine.